Add mapping of untyped nil args to `BindNull` to exec docs

davidjb opened the pull request:
In my app, values passed in from JSON (and elsewhere) where they could be a value or `NULL` in SQL are stored as typed pointers, such as `*string`. In order to ensure these are handled by `sqlitex` and bound as `BindNull`, the `nil` value needs to be passed as an untyped nil, rather than a `*string`

This needs to be handled on my side so this PR documents the handling of untyped nil to BindNull in the `ExecOptions` documentation, alongside how other types are handled.

This also slightly adjusts the default handling documentation, so each section mentions `fmt.Sprint(v)` and aligns with the implementation (https://github.com/zombiezen/go-sqlite/blob/main/sqlitex/exec.go#L327).

zombiezen commented on sqlitex/exec.go:
```diff
@@ -35,11 +35,12 @@ type ExecOptions struct {
-	//	[]byte   to BindBytes
-	//	string   to BindText
-	//	bool     to BindBool
+	//	integers    to BindInt64
```
Let's turn these into [lists](https://go.dev/doc/comment#lists) and remove the special spacing.

davidjb replied:
For sure 👍  GitHub's UI editing swapped the original tabs for spaces - reasons not to trust it. I've opted to match the `sqlitex/pool.go` syntax for lists, the `sqlite.go` comment syntax has an extra space indent.

zombiezen commented on sqlitex/exec.go:
```diff
@@ -76,13 +78,14 @@ type ExecOptions struct {
+//      untyped nil to BindNull
 //
-// All other kinds are printed using fmt.Sprintf("%v", v) and passed
+// All other kinds are printed using fmt.Sprint(v) and passed
```
nit: And use a [link](https://go.dev/doc/comment#doclinks) here:

```suggestion
// All other kinds are printed using [fmt.Sprint] and passed
```

zombiezen reviewed (commented):
Sorry to leave you hanging for so long. I'm not sure how this slipped through the cracks on my end. A few minor tweaks, please:

davidjb commented:
Thanks for the review @zombiezen. I've updated the PR and fixed the inconsistent spacing (tabs/spaces). This is what it looks like now when rendered:

<img width="75%" alt="rendering of sqlitex docs after formatting changes" src="https://github.com/user-attachments/assets/dacacfc9-614c-4ef4-9b56-4385923f21c7" />

<img width="75%" alt="rendering of sqlitex Exec docs after formatting changes" src="https://github.com/user-attachments/assets/5c2ea797-74b2-42b2-85fc-5eda5f60e3a6" />

I updated the `fmt.Sprintf` references in the type struct comments to match the `Exec` docs; the latter renders as links but the former doesn't so let me know your thoughts. 

Let me know if any thing else needs adjusting.

zombiezen reviewed (approved):
LGTM! Thanks.
